cpp/src/arrow/testing/gtest_util.h

```diff
@@ -309,6 +309,9 @@ ARROW_TESTING_EXPORT void ApproxCompareBatch(
     const RecordBatch& left, const RecordBatch& right, bool compare_metadata = true,
     const EqualOptions& options = TestingEqualOptions());
 
+ARROW_TESTING_EXPORT void RoundtripBatch(const std::shared_ptr<RecordBatch>& batch,
+                                         std::shared_ptr<RecordBatch>* out);
```

Comment:
Since this depends on IPC, I would move it into `arrow/ipc/test_common.{h,cc}`.

Reply:
Moved.